### PR TITLE
Companies stat

### PR DESCRIPTION
**Pull request by anpolimus**
1. Users sync batch now also updates existed users.
2. Home page most widget includes companies info to stat + d.org direct links

### CODE REVIEW

**Comment by anpolimus** [comment]
@podarok needs review.

File: docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module [hunk 1]
```diff
@@ -542,7 +574,7 @@ function ppgetstat_node_insert($node) {
   if (!isset($node->field_user_id[LANGUAGE_NONE][0]['value'])) {
     return;
   }
-  $queue = DrupalQueue::get('stats_jobs');
+  /*$queue = DrupalQueue::get('stats_jobs');
```
Comment (podarok) [inline docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module]: We need to create views bulk operation then for ability to add users to scanning queue

File: docroot/sites/all/modules/custom/pp_frontpage/plugins/content_types/most.inc [hunk 2]
```diff
@@ -33,26 +33,44 @@ function pp_frontpage_most_content_type_render($subtype, $conf, $panel_args, $co
...
+      $user_node = $user_nodes[$doid];
+      if (!empty($user_node->field_organisations)) {
+        $company_titles = array();
+        $companies = field_get_items('node', $user_node, 'field_organisations');
+        foreach ($companies as $company) {
+          $company_titles[] = taxonomy_term_load($company['tid'])->name;
+        }
+        $company_column = implode(',', $company_titles);
+      } else {
+        $company_column = '';
```
Comment (podarok) [inline docroot/sites/all/modules/custom/pp_frontpage/plugins/content_types/most.inc]: Use Early Define pattern, please https://github.com/propeoplemd/cibox/wiki/PHP-and-Drupal#define-early-pattern

File: docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module [hunk 3]
```diff
@@ -251,6 +251,38 @@ function _ppgetstats_get_all_users() {
   return array();
 }
 
+/**
+ * Helper function that gets current project ids by d.org ids.
+ * @param $doids
```
Comment (podarok) [inline docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module]: Should be a type for parameter descibed as well

File: docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module [hunk 4]
```diff
@@ -251,6 +251,38 @@ function _ppgetstats_get_all_users() {
...
+ */
+function _ppgetstat_get_users_by_doids($doids) {
+  $query = new EntityFieldQuery();
```
Comment (podarok) [inline docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module]: move to line 263, please

File: docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module [hunk 5]
```diff
@@ -251,6 +251,38 @@ function _ppgetstats_get_all_users() {
   return array();
 }
 
+/**
+ * Helper function that gets current project ids by d.org ids.
```
Comment (podarok) [inline docroot/sites/all/modules/custom/ppgetstat/ppgetstat.module]: missed @return docblock's part

File: docroot/sites/all/modules/custom/ppgetstat/src/DrupalUserTrackingCode.php [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+      $this->ids->next();
+      $this->getCount($this->nextUpId($id));
+    }
+    else {
+      $this->ids->offsetSet($id, FALSE);
+      $this->getCount($this->nextDownId($id));
+    }
+  }
+
+  private function nextUpId($current = 0) {
```
Comment (podarok) [inline docroot/sites/all/modules/custom/ppgetstat/src/DrupalUserTrackingCode.php]: doblock is missed

**Comment by anpolimus** [comment]
@podarok, fixed + implemented ability to update single user by DOID

**Pull request merged**